mono/mini/aot-compiler.c

```diff
@@ -7509,7 +7544,7 @@ compile_method (MonoAotCompile *acfg, MonoMethod *method)
 		if (acfg->aot_opts.print_skipped_methods)
 			printf ("Skip (abs call): %s\n", mono_method_get_full_name (method));
 		InterlockedIncrement (&acfg->stats.abscount);
-		mono_destroy_compile (cfg);
+		pending_compile_destroy (acfg, cfg);
 		return;
```

Comment:
These are failed compiles, but the types that these refer to are freed by mono_destroy_compile. If we free them earlier, we end up with memory unsafety. Currently we don't free them later I'm pretty sure, we leak them. I'm going to figure out why we're not reparenting the types though, we should be able to promptly free the memory.